### PR TITLE
gateway-api: Support latest release v1.2.0

### PR DESCRIPTION
**Pull request by sayboras**
Relates: https://github.com/kubernetes-sigs/gateway-api/releases/tag/v1.2.0

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -385,9 +385,9 @@ and the `2-Clause BSD License <bsd-license_>`__
     :alt: FOSSA Status
     :target: https://app.fossa.com/projects/custom%2B162%2Fgit%40github.com%3Acilium%2Fcilium.git?ref=badge_shield
 
-.. |gateway-api| image:: https://img.shields.io/badge/Gateway%20API%20Conformance%20v1.0.0-Cilium-green
+.. |gateway-api| image:: https://img.shields.io/badge/Gateway%20API%20Conformance%20v1.2.0-Cilium-green
     :alt: Gateway API Status
-    :target: https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/reports/v1.0.0/cilium.yaml
+    :target: https://github.com/kubernetes-sigs/gateway-api/tree/main/conformance/reports/v1.2.0/cilium-cilium/
```
Comment (joestringer) [inline README.rst]: This target is generating a 404, but I guess once the conformance report goes through it should be at this URL?

https://github.com/kubernetes-sigs/gateway-api/tree/main/conformance/reports/v1.2.0/cilium-cilium/

Comment (joestringer) [reply README.rst]: Ah yes I see https://github.com/kubernetes-sigs/gateway-api/pull/3372 .

**Review by qmonnet — APPROVED** [review]

**Review by marseel — APPROVED** [review]

**Pull request merged**